### PR TITLE
Fixes #688 - Adds Products bulk actions support.

### LINKED ISSUE #688: Add Products bulk actions support.

### PR DESCRIPTION
**Pull request by jameerpathan111**
Fixes #688 

Test result:
```
In [2]: entities.ProductsBulkActions().destroy(data={"ids":[63,64]})                                                                                                                       
Out[2]: 
{'displayMessages': {'success': ['Successfully initiated removal of 2 product(s)'],
  'error': []}}

In [4]: entities.ProductsBulkActions().sync_plan(data={"ids":[59,58,61], "plan_id":1})                                                                                                     
Out[4]: 
{'displayMessages': {'success': ['Successfully changed sync plan for 3 product(s)'],

In [9]: entities.ProductsBulkActions().http_proxy(data={"ids":[1], "http_proxy_policy":"use_selected_http_proxy", "http_proxy_id":1})                                                      
Out[9]: 
{'id': 'a454386e-5155-47af-a758-26736ebf5e47',
 'label': 'Actions::Katello::Product::UpdateHttpProxy',
 'pending': False,
 'action': 'Update http Capsule',
 'username': 'admin',
 'started_at': '2019-12-17 13:00:35 UTC',
 'ended_at': '2019-12-17 13:00:35 UTC',
 'state': 'stopped',
 'result': 'success',
 'progress': 1.0,
 'input': {'current_request_id': None,
  'current_timezone': 'UTC',
  'current_user_id': 4,
  'current_organization_id': None,
  'current_location_id': None},
 'output': {},
 'humanized': {'action': 'Update http Capsule',
  'input': [],
  'output': '',
  'errors': []},
 'cli_example': None,
 'available_actions': {'cancellable': False, 'resumable': False}}
```

### CODE REVIEW

**Comment by jameerpathan111** [comment]
Test result: 
```
In [2]: entities.Repository(id=9, http_proxy_policy='none').update()                                                                                                                       
Out[2]: nailgun.entities.Repository(backend_identifier='daf2ce59-9947-493e-b6c6-580de5b805bb', checksum_type=None, content_counts={'ostree_branch': 0, 'docker_manifest': 0, 'docker_manifest_list': 0, 'docker_tag': 0, 'rpm': 62, 'srpm': 0, 'package': 62, 'package_group': 1, 'erratum': 0, 'puppet_module': 0, 'file': 0, 'deb': 0, 'module_stream': 0, 'ansible_collection': 0}, content_type='yum', container_repository_name=None, docker_upstream_name=None, docker_tags_whitelist=None, download_policy='on_demand', full_path='http://hostname.example.com/pulp/repos/Default_Organization/Library/custom/Sat6Tools7/sat6tool7/', gpg_key=None, ignorable_content=None, label='sat6tool7', last_sync=nailgun.entities.ForemanTask(id='6b30413c-6e47-4063-926b-cf2b6de3db6a'), mirror_on_sync=True, name='sat6tool7', product=nailgun.entities.Product(id=59), unprotected=True, url='http://content.example.com/pulp/repos/Sat6-CI/QA/Tools_6_7_with_RHEL7_Server/custom/Satellite_Tools_6_7_Composes/Satellite_Tools_6_7_RHEL7_x86_64/', upstream_username=None, verify_ssl_on_sync=True, http_proxy_policy='none', http_proxy_id=None, id=9)

In [3]: entities.Repository(id=9, http_proxy_policy='use_selected_http_proxy', http_proxy_id=1).update()                                                                                   
Out[3]: nailgun.entities.Repository(backend_identifier='daf2ce59-9947-493e-b6c6-580de5b805bb', checksum_type=None, content_counts={'ostree_branch': 0, 'docker_manifest': 0, 'docker_manifest_list': 0, 'docker_tag': 0, 'rpm': 62, 'srpm': 0, 'package': 62, 'package_group': 1, 'erratum': 0, 'puppet_module': 0, 'file': 0, 'deb': 0, 'module_stream': 0, 'ansible_collection': 0}, content_type='yum', container_repository_name=None, docker_upstream_name=None, docker_tags_whitelist=None, download_policy='on_demand', full_path='http://hostname.example.com/pulp/repos/Default_Organization/Library/custom/Sat6Tools7/sat6tool7/', gpg_key=None, ignorable_content=None, label='sat6tool7', last_sync=nailgun.entities.ForemanTask(id='6b30413c-6e47-4063-926b-cf2b6de3db6a'), mirror_on_sync=True, name='sat6tool7', product=nailgun.entities.Product(id=59), unprotected=True, url='http://content.example.com/pulp/repos/Sat6-CI/QA/Tools_6_7_with_RHEL7_Server/custom/Satellite_Tools_6_7_Composes/Satellite_Tools_6_7_RHEL7_x86_64/', upstream_username=None, verify_ssl_on_sync=True, http_proxy_policy='use_selected_http_proxy', http_proxy_id=1, id=9)

In [6]: entities.Repository(name="test", url="https://example.com", product=59, organization=1, content_type="yum", http_proxy_policy="use_selected_http_proxy", http_proxy_id=1).create() 
Out[6]: nailgun.entities.Repository(backend_identifier='bae286b6-5cf9-4769-8e0b-51cc4eaca248', checksum_type=None, content_counts={'ostree_branch': 0, 'docker_manifest': 0, 'docker_manifest_list': 0, 'docker_tag': 0, 'rpm': 0, 'srpm': 0, 'package': 0, 'package_group': 0, 'erratum': 0, 'puppet_module': 0, 'file': 0, 'deb': 0, 'module_stream': 0, 'ansible_collection': 0}, content_type='yum', container_repository_name=None, docker_upstream_name=None, docker_tags_whitelist=None, download_policy='on_demand', full_path='http://hostname.example.com/pulp/repos/Default_Organization/Library/custom/Sat6Tools7/test/', gpg_key=None, ignorable_content=None, label='test', last_sync=None, mirror_on_sync=True, name='test', product=nailgun.entities.Product(id=59), unprotected=True, url='https://example.com', upstream_username=None, verify_ssl_on_sync=True, http_proxy_policy='use_selected_http_proxy', http_proxy_id=1, id=54)

In [7]: entities.Repository(id=54).delete()                                                                                                                                                
Out[7]: {}
```

**Comment by jameerpathan111** [comment]
@vijay8451 add http_proxy repository support as well.

**Comment by ntkathole** [comment]
ok to test : tests/foreman/api/test_repository.py -m tier1

**Comment by vijay8451** [comment]
Failures tests are not related to new change ... hence looks good to me .

File: nailgun/entities.py [hunk 1]
```diff
@@ -5467,6 +5467,114 @@ def sync(self, synchronous=True, **kwargs):
         return _handle_response(response, self._server_config, synchronous)
 
 
+class ProductsBulkActions(Entity):
```
Comment (ntkathole) [inline nailgun/entities.py]: class ProductsBulkAction(Entity):
all class names in nailgun has to be singular

File: nailgun/entities.py [hunk 2]
```diff
@@ -5467,6 +5467,114 @@ def sync(self, synchronous=True, **kwargs):
...
+    """A representation of a Products bulk actions entity."""
+
+    def __init__(self, server_config=None, **kwargs):
+        self._meta = {
+            'api_path': '/katello/api/products/bulk',
+            'server_modes': ('sat', 'sam'),
```
Comment (ntkathole) [inline nailgun/entities.py]: no need of adding server_modes as we are not using it now.

File: nailgun/entities.py (hunk 1 above)
Comment (jameerpathan111) [reply nailgun/entities.py]: @ntkathole updated PR.

File: nailgun/entities.py (hunk 2 above)
Comment (jameerpathan111) [reply nailgun/entities.py]: updated PR.

File: nailgun/entities.py (hunk 1 above)
Comment (ntkathole) [reply nailgun/entities.py]: @jameerpathan111 s/Actions/Action/

Comment (jameerpathan111) [reply nailgun/entities.py]: updated PR.

**Pull request merged**